endpoints/openrtb2/auction_test.go

```diff
@@ -5156,46 +5248,112 @@ func (e mockStageExecutor) GetOutcomes() []hookexecution.StageOutcome {
 
 func TestSetSeatNonBidRaw(t *testing.T) {
 	type args struct {
-		request         *openrtb_ext.RequestWrapper
-		auctionResponse *exchange.AuctionResponse
+		request  *openrtb_ext.RequestWrapper
+		response *openrtb2.BidResponse
+		nonBids  []openrtb_ext.SeatNonBid
+	}
+	type want struct {
+		error    bool
+		response *openrtb2.BidResponse
 	}
 	tests := []struct {
-		name    string
-		args    args
-		wantErr bool
+		name string
+		args args
+		want want
 	}{
 		{
-			name:    "nil-auctionResponse",
-			args:    args{auctionResponse: nil},
-			wantErr: false,
+			name: "nil response",
+			args: args{response: nil},
+			want: want{
+				error:    false,
+				response: nil,
+			},
 		},
 		{
-			name:    "nil-bidResponse",
-			args:    args{auctionResponse: &exchange.AuctionResponse{BidResponse: nil}},
-			wantErr: false,
+			name: "returnallbidstatus false",
+			args: args{response: &openrtb2.BidResponse{},
+				request: &openrtb_ext.RequestWrapper{BidRequest: &openrtb2.BidRequest{Ext: []byte(`{"prebid": { "returnallbidstatus" : false }}`)}}},
+			want: want{
+				error:    false,
+				response: &openrtb2.BidResponse{},
+			},
 		},
 		{
-			name:    "invalid-response.Ext",
-			args:    args{auctionResponse: &exchange.AuctionResponse{BidResponse: &openrtb2.BidResponse{Ext: []byte(`invalid_json`)}}},
-			wantErr: true,
+			name: "invalid responseExt",
+			args: args{
+				request:  &openrtb_ext.RequestWrapper{BidRequest: &openrtb2.BidRequest{Ext: []byte(`{"prebid": { "returnallbidstatus" : true }}`)}},
+				response: &openrtb2.BidResponse{Ext: []byte(`{invalid}`)},
+				nonBids: []openrtb_ext.SeatNonBid{
+					{
+						Seat: "pubmatic",
+						NonBid: []openrtb_ext.NonBid{
+							{
+								ImpId:      "imp",
+								StatusCode: 1,
+							},
+						},
+					},
+				},
+			},
+			want: want{
+				error:    true,
+				response: &openrtb2.BidResponse{Ext: []byte(`{invalid}`)},
+			},
+		},
+		{
+			name: "returnallbistatus is true, update seatnonbid in nil responseExt",
```

Comment:
typo in the name, should be "returnallbi`d`status is true..."

Reply:
Fixed.